### PR TITLE
GODRIVER-1349 implement mgocompat StructCodec options

### PR DESCRIPTION
**Pull request by iwysiu**
The lax decoding work was removed from the scope of this ticket, as it is impossible to add an option for mgobson's specific lax decoding behavior without backwards breaking changes.

### CODE REVIEW

File: bson/bsoncodec/struct_codec.go [hunk 1]
```diff
@@ -257,6 +283,23 @@ func (sc *StructCodec) isZero(i interface{}) bool {
 		return v.Float() == 0
 	case reflect.Interface, reflect.Ptr:
 		return v.IsNil()
+	case reflect.Struct:
+		if sc.OmitDefaultStruct {
```
Comment (craiggwilson) [inline bson/bsoncodec/struct_codec.go]: Does having this here change the current behavior? It seems like if we don't enter this, we are always returning false.

I guess by the naming of this, that is inevitable, but it's unclear as to whether current users are wanting this.

Comment (iwysiu) [reply bson/bsoncodec/struct_codec.go]: Originally isZero always returned false for structs that don't implement the Zeroer interface, so the default behavior is the same.

Comment (craiggwilson) [reply bson/bsoncodec/struct_codec.go]: works for me. :)

File: bson/bsonoptions/struct_codec_options.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	return t
+}
+
+// SetDecodeDeepZeroInline specifies if structs should be zeroed before decoding into them. Defaults to false.
+func (t *StructCodecOptions) SetDecodeDeepZeroInline(b bool) *StructCodecOptions {
+	t.DecodeDeepZeroInline = &b
+	return t
+}
+
+// SetOmitDefaultStruct specifies if default structs should be considered empty by omitempty. Defaults to false.
```
Comment (divjotarora) [inline bson/bsonoptions/struct_codec_options.go]: 1. Explain what a "default" struct is in the comments.
2. Is `omitempty` relevant to both encoding and decoding? If it's only relevant to one of them, the option name should include `Encode` or `Decode`.

**Pull request merged**